terraform/ec2-instances-creator/main.tf

```diff
@@ -0,0 +1,55 @@
+locals {
+  # Available fields in each element:
+  #  {
+  #    "fbVersion": "2.0.7",
+  #    "ec2User": "centos",
+  #    "osDistro": "centos",
+  #    "osVersion": 9,
+  #    "arch": "x86_64",
+  #    "ami": "ami-08c92aec9ccf0e1e9",
+  #    ...
+  #  }
+  instance_matrix = jsondecode(file(var.instance_matrix_file))
+
+  aws_vpc_subnet = "subnet-06240c469195932cf"
+  ec2_instances_security_group = "sg-0c4318be91bbe3cba"
+  ec2_instance_profile = "logging-e2e-testing-ec2-ssm-instance-profile"
```

Comment:
We need to pass this instance profile to allow the EC2 instances to connect to AWS SSM. This instance profile has been created on the backend infra with the minimum required rights.